tianshou/data/batch.py

```diff
@@ -433,11 +437,11 @@ def __init__(
             # Feels like kwargs could be just merged into batch_dict in the beginning
             self.__init__(kwargs, copy=copy)  # type: ignore
 
-    def to_dict(self) -> dict[str, Any]:
+    def to_dict(self, recurse: bool = False) -> dict[str, Any]:
```

Comment:
I think recursive is better, feel free to rename `recurse`  to `recursive` wherever the former is used

Reply:
Btw, I had a look at the empty implementation, which seems very weird to me. I don't feel like the `recurse=False` option should exist at all. A user can always just check for lens of the keys instead. Moreover, the recursive=True corresponds to checking for `len` of the batch.

I feel like `is_empty` can be removed completely. If you agree, no need to rename the `recurse` - I'd set up an issue for remove `is_empty`

Reply:
In fact, adding this issue right now, since it's good for newcomers

Reply:
`Batch.to_torch()` has same problem because it is also in-place. We can change this in another PR?

Reply:
Could you change to `recursive` pls? Then this is good to go, pls just add the changes to changelog and the PR description.

Again, thanks for the work on this, highly appreciated!